pkg/cmd/cli/cmd/expose.go

```diff
@@ -132,7 +132,7 @@ func validate(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
 			cmd.Flags().Set("generator", generator)
 			fallthrough
 		case "route/v1":
-			route, err := cmdutil.UnsecuredRoute(kc, namespace, info.Name, info.Name, kcmdutil.GetFlagString(cmd, "port"))
+			route, err := cmdutil.UnsecuredRoute(kc, namespace, info.Name, info.Name, kcmdutil.GetFlagString(cmd, "port"), true)
```

Comment:
You need to explain why useTargetPort is set to true here.  Someone looking at this in the future will have no idea that the upstream generator will set the route port to he service port (rather than target port).